system-addon/lib/TopSitesFeed.jsm

```diff
@@ -88,8 +88,8 @@ this.TopSitesFeed = class TopSitesFeed {
   }
 
   async getLinksWithDefaults() {
-    // Get at least 2 rows so toggling between 1 and 2 rows has sites
-    const numItems = Math.max(this.store.getState().Prefs.values.topSitesRows, 2) * TOP_SITES_MAX_SITES_PER_ROW;
+    // Get at least 4 rows so toggling between 1, 2, 3, 4 rows has sites
+    const numItems = Math.max(this.store.getState().Prefs.values.topSitesRows, 4) * TOP_SITES_MAX_SITES_PER_ROW;
```

Comment:
I guess we could only fetch what we need and then refresh whenever the pref is changed. That might be better now that we go up to 4?

Reply:
Pretty sure this will cause a lot of extra storage, e.g,. thumbnails and icons, if the user only has the default 1 row. Perhaps now that adding/removing rows is done through a separate about:preferences page, a slight delay when it updates in the background off screen, it won't be an issue of not having an immediately responsive toggling.

Reply:
Yep yep. I think we were more worried about that when preferences was the in-page side panel.